File: files/2025-09-21/CSEC 21092025.R
library(tidyverse)
library(readxl)

path = "files/2025-09-21/Challenge 61.xlsx"
input = read_excel(path, range = "B2:F8")
test  = read_excel(path, range = "H2:J8")

# short but little bit difficult
result = input %>%
  select(-where(~all(is.na(.)))
         
# short and easy
result2 = input %>%
  janitor::remove_empty("cols")

all.equal(test, result, check.attributes = FALSE) # True
all.equal(test, result2, check.attributes = FALSE) # True

File: files/2025-09-21/CSEC 21092025.py
import pandas as pd

path = "files/2025-09-21/Challenge 61.xlsx"
input = pd.read_excel(path, usecols="B:F", skiprows=1, nrows=6)
test = pd.read_excel(path, usecols="H:J", skiprows=1, nrows=6).rename(columns=lambda x: x.replace('.1', ''))

result = input.dropna(axis=1, how='all')
print(test.equals(result))   # True

result2 = input.loc[:, input.notna().any()]
print(test.equals(result2))  # True
